sci-libs/openblas/openblas-0.3.0.recipe

```diff
@@ -0,0 +1,83 @@
+SUMMARY="An optimized BLAS library"
+DESCRIPTION="OpenBLAS is an optimized BLAS library based on GotoBLAS2 1.13 BSD\
+ version."
+HOMEPAGE="http://www.openblas.net/"
+COPYRIGHT="2011-2018 The OpenBLAS Project"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="http://github.com/xianyi/OpenBLAS/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="cf51543709abe364d8ecfb5c09a2b533d2b725ea1a66f203509b21a8e9d8f1a1"
+SOURCE_DIR="OpenBLAS-$portVersion"
+PATCHES="openblas-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="!x86_gcc2 x86"
+
+PROVIDES="
+	openblas$secondaryArchSuffix = $portVersion
+	lib:libopenblas$secondaryArchSuffix = $portVersion compat >= 0
+	lib:libopenblasp_r${portVersion}$secondaryArchSuffix = $portVersion compat >= 0
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libgfortran$secondaryArchSuffix
+	lib:libgomp$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	openblas${secondaryArchSuffix}_devel = $portVersion
+	devel:libopenblas$secondaryArchSuffix = $portVersion compat >= 0
+	devel:libopenblasp_r${portVersion}$secondaryArchSuffix = $portVersion compat >= 0
+	"
+REQUIRES_devel="
+	openblas$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:gfortran$secondaryArchSuffix
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	cmd:perl
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage openblas$secondaryArchSuffix \
+	"$libDir"/libopenblas.so.0 \
+	"$libDir"/libopenblasp-r${portVersion}.so
+
+BUILD()
+{
+	# DYNAMIC_ARCH is x86 only
+	# PPC & others should specify TARGET=
+	target=
+	[ "$targetArchitecture" == x86 ] && target=NORTHWOOD
+	[ "$targetArchitecture" == x86_64 ] && target=PRESCOTT
```

Comment:
```diff
-	target=
-	[ "$targetArchitecture" == x86 ] && target=NORTHWOOD
-	[ "$targetArchitecture" == x86_64 ] && target=PRESCOTT
+	case "$effectiveTargetArchitecture" in
+		x86) target=NORTHWOOD;;
+		x86_64) target=PRESCOTT;;
+		*) target=;;
+	esac
```

Without this change, `BUILD()` would fail as soon as it reaches a test that returns false.